tests/integrationv2/test_signature_algorithms.py

```diff
@@ -176,7 +188,66 @@ def test_s2n_client_signature_algorithms(managed_process, cipher, provider, othe
-        assert signature_marker(
-            Provider.ServerMode, signature) in results.stdout or not server_sigalg_used
-        assert (signature_marker(Provider.ClientMode, signature)
-                in results.stdout) == client_auth
+        # TLS1.1 always negotiates to RSA+MD5_SHA1 signature algorithm.
+        if protocol < Protocols.TLS12:
+            assert signature_marker(
+                Provider.ServerMode, Signatures.MD5_SHA1
+            ) in results.stdout
+            assert (signature_marker(Provider.ClientMode, Signatures.MD5_SHA1)
+                    in results.stdout) == client_auth
+        else:
+            assert signature_marker(
+                Provider.ServerMode, signature) in results.stdout or not server_sigalg_used
+            assert (signature_marker(Provider.ClientMode, signature)
+                    in results.stdout) == client_auth
+
+
+@pytest.mark.uncollect_if(func=skip_ciphers)
+@pytest.mark.parametrize("cipher", [Ciphers.DEFAULT_FIPS], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("protocol", [Protocols.TLS12], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("signature", sig_list, ids=get_parameter_name)
+@pytest.mark.parametrize("client_auth", [True, False], ids=lambda val: "client-auth" if val else "no-client-auth")
+def test_s2n_server_signature_algorithms_tls12_behavior(managed_process, cipher, provider, protocol, certificate,
+                                              signature, client_auth):
+    port = next(available_ports)
+
+    random_bytes = data_bytes(64)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        data_to_send=random_bytes,
+        insecure=False,
+        signature_algorithm=signature,
+        protocol=protocol
+    )
+
+    server_options = copy.copy(client_options)
+    server_options.extra_flags = None
+    server_options.data_to_send = None
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+    server_options.cipher = cipher
+
+    server = managed_process(S2N, server_options, timeout=5)
+    client = managed_process(provider, client_options, timeout=5)
+
+    for results in client.get_results():
+        results.assert_success()
+
+    expected_version = get_expected_s2n_version(protocol, provider)
+
+    for results in server.get_results():
+        results.assert_success()
+        assert to_bytes("Actual protocol version: {}".format(
+            expected_version)) in results.stdout
+
+        # As per TLS1.2, if the negotiated key exchange algorithm is one of (RSA, DHE_RSA,
+        # DH_RSA, RSA_PSK, ECDH_RSA, ECDHE_RSA), server is expected to behave as if client had sent value {sha1,rsa}.
+        #
+        # This is inferred from the rfc- https://www.rfc-editor.org/rfc/rfc5246#section-7.4.1.4.1
```

Comment:
It might be good to make it clear here that you're causing a signature algorithm mismatch, and that s2n-tls invokes this fallback behavior upon signature algorithm mismatch as well as not receiving a signature_algorithms extension (since this RFC section just says what to do if the extension wasn't sent).